### PR TITLE
Added Pylint and code cleanup

### PR DESCRIPTION
**Pull request by gaurav**
Added Pylint to standardize code style and changed the code to fix all Pylint errors. Also added Pylint to Travis so that code style will continue to be tested in the future.

I also tried to improve the [phyloreference documentation](https://github.com/gaurav/phylo2owl/blob/79466a9edfce2642feeef3afa389326d0b031470/examples/trees/journal.pone.0094199.md). This work will be continued in phyloref/curation-tool#64.

### CODE REVIEW

**Review by hlapp — COMMENTED** [review]
As for the style changes in `journal.pone.0094199.md`, what I had in mind was something a little more subtle. For example, leaving the property and entity identifiers in normal font and changing OWL reserved syntax words (_value_, _and_, _some_, etc) to italics.

In general I think we want to (1) maintain easy readability   while making identifiers easily distinguishable from OWL reserved word without the distinction visually taking center stage; and (2) start introducing a convention that will carry over well into publications so we don't have to use a different convention there.

**Comment by gaurav** [comment]
@hlapp Done! Check it out at https://github.com/gaurav/phylo2owl/blob/79466a9edfce2642feeef3afa389326d0b031470/examples/trees/journal.pone.0094199.md (take 2)

I think the OWL keywords look pretty good now, but I'd like the names to be more immediately recognizable as names-in-OWL, rather than just words. I tried putting ':' in front of them, but I think that just looks annoying: https://github.com/gaurav/phylo2owl/blob/c22fbb1bf47acc3eb5e45551be23c9ecb186cf36/examples/trees/journal.pone.0094199.md (take 3). So if I had to choose, take 2 is probably the best representation for right now.

We could potentially turn those names into URLs, which on browsers would make them visually distinct and would allow someone to immediately look up "excludes_lineage_to" and other terms. Do you think that'd be useful?

**Comment by hlapp** [comment]
> We could potentially turn those names into URLs, which on browsers would make them visually distinct and would allow someone to immediately look up "excludes_lineage_to" and other terms. Do you think that'd be useful?

Yes, but that shouldn't be manual should it? Or what way to do this did you have in mind?

**Comment by hlapp** [comment]
@gaurav where are we at with this?

**Comment by gaurav** [comment]
This pull request covers two things:
1. How to display phyloreferences [in OWL Manchester syntax](https://github.com/gaurav/phylo2owl/blob/79466a9edfce2642feeef3afa389326d0b031470/examples/trees/journal.pone.0094199.md). This is largely deprecated by PHYX now: it's better to describe phyloreferences in terms of specifiers rather than as the OWL property restriction representing them. However, I've opened an issue so that users can view the OWL property restriction in the Curation Tool if they want to (phyloref/curation-tool#64).
2. A minor code cleanup. Based on our experience with the Curation Tool, I've added [pylint](https://www.pylint.org/) to our test suite and fixed all major Pylint errors. This is what this pull request primarily covers now, so I'll change its description to match that.

I forgot about this pull request because we've put phylo2owl on hold until we can replace it with the Python library being developed as part of the [Clade Ontology](https://github.com/phyloref/clade-ontology). But thanks for flagging this pull request up -- once it's been reviewed and merged, phylo2owl can go properly on ice until it's time to work on it again.

Finally, this was from the bad old days when I would develop in my own fork before pushing to phyloref. I've moved all the other branches into the phyloref repository now. I'll keep this branch on my own fork until it's reviewed and merged, and then I'll delete my fork.

File: phylo2owl.py
Comment (hlapp) [inline phylo2owl.py]: Why not simply _Convert phylogenies into OWL ontologies_? That the OWL is serialized in RDF/XML seems more like current and likely temporary limitation that something important to its design or purpose.

File: phylo2owl.py
Comment (hlapp) [inline phylo2owl.py]: See comment above.

File: tests/conftest.py
Comment (hlapp) [inline tests/conftest.py]: This is guarantees failing cross-platform compatibility. Isn't there a constant for for path separator, or a function that joins paths in the manner appropriate for the OS it's running on?

File: tests/test_alt_inputs.py
Comment (hlapp) [inline tests/test_alt_inputs.py]: Can we add what this means, beyond an impenetrable code?

**Review by hlapp — APPROVED** [review]
Only minor in-line comments.

**Pull request merged**
